.github/workflows/main.yml

```diff
@@ -0,0 +1,100 @@
+name: ci
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        python-version:
+          - 3.8
+          - 3.6
+    
+    services:
+      postgres:
+        image: postgres:13
+        env:
+          POSTGRES_PASSWORD: postgres
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: >-
+          sudo apt-get install -y
+          build-essential
+          php7.4-cli
+          haskell-platform
+          fp-compiler
+          python2.7
+          cppreference-doc-en-html
+          cgroup-lite
+          libcap-dev
+          python-dev
+          libpq-dev
+          libcups2-dev
+          libyaml-dev
+          rustc
+          mono-mcs
+
+      - name: Install requirements
+        run: |
+          pip install -r requirements.txt
+          pip install -r dev-requirements.txt
+
+          sudo $(which python) prerequisites.py -y --cmsuser=$USER install
+          python setup.py install
+
+      - name: Change group for working and temp directories to $USER
+        run: |
+          sudo chgrp $USER . /tmp
+          sudo chmod g+s . /tmp
+
+      - name: Change user for isolate to root
+        run: |
+          sudo chown root:$USER ./isolate/isolate
+          sudo chmod u+s ./isolate/isolate
+
+      - name: Set up database
+        env:
+          PGPASSWORD: postgres
+        run: |
+          psql --host=localhost --username=postgres --command="CREATE USER cmsuser WITH PASSWORD 'your_password_here';"
+
+          createdb --host=localhost --username=postgres --owner=cmsuser cmsdb
+          psql --host=localhost --username=postgres --dbname=cmsdb --command='ALTER SCHEMA public OWNER TO cmsuser'
+          psql --host=localhost --username=postgres --dbname=cmsdb --command='GRANT SELECT ON pg_largeobject TO cmsuser'
+
+          createdb --host=localhost --username=postgres --owner=cmsuser cmsdbfortesting
+          psql --host=localhost --username=postgres --dbname=cmsdbfortesting --command='ALTER SCHEMA public OWNER TO cmsuser'
+          psql --host=localhost --username=postgres --dbname=cmsdbfortesting --command='GRANT SELECT ON pg_largeobject TO cmsuser'
+
+          cp ./config/cms.conf.sample ./config/cms.conf
+          sed -e "s/\"cmsuser\": \"cmsuser\"/\"cmsuser\": \"$USER\"/" -i ./config/cms.conf
+
+          cmsInitDB
+
+      - name: Create cache and log directories for tests
+        run: | 
+          mkdir cache log
```

Comment:
Need to create this in advance, otherwise the test will fail with permission error on this line:

https://github.com/cms-dev/cms/blob/master/cms/util.py#L60

I'm suspecting this is because the default group in GitHub Actions is `docker`. Do not have bandwidth to investigate further. Please advise if there is better approach.